### DISCUSSION TITLE
App build slow / unresponsive on the phone

### DISCUSSION THREAD

**Discussion by i-mage** [quasarframework/quasar#9666] [CLI - Capacitor / Cordova modes]
I am developing an app which uses combination of sounds / animations & everything was great and smooth once developing on the notebook and DEV mode on the phone. However once I have build the app in production mode, suddenly the app started lagging and being unresponsive. 

Could anyone pls help me how to optimize the app or increase the performance?

**Comment by ontwikkelfabriek** [comment]
@i-mage 

>  However once I have build the app in production mode, suddenly the app started lagging and being unresponsive.

Is your app running slow in 'prod' on the **same** devices that run it good in 'dev'?

**Reply by i-mage** [reply]
Yes, exactly the same device - iPhone 7, iOS 14.4

**Reply by ontwikkelfabriek** [reply]
@i-mage 

> Yes, exactly the same device - iPhone 7, iOS 14.4

I would first try to run it on another device and see what it does. Other than that I would **not** be looking for _"optimize the app or increase the performance"_  in a general way , that's because it runs without lag in dev mode, it should be the other way around, dev is inherently slower than prod builds.  You should be looking for **why** it is running slowly on **prod** builds, other than that I cannot be of much more help without any other clues.

**Reply by i-mage** [reply]
Thanks for the insights! I will proceed as follows:

- try different devices, I have several iPhones at home at my disposal
- in case it won't help, I will try to minimize the code & proceed section by section

Do you think there's a chance it could be an issue with XCode version? E.g. should I update to the last stable version?

**Reply by ontwikkelfabriek** [reply]
@i-mage 

> try different devices, I have several iPhones at home at my disposal

If possible try it on an Android device too.

> Do you think there's a chance it could be an issue with XCode version? E.g. should I update to the last stable version?

Unlikely....but you can always try. 

Xcode probably also has profiling tools:
https://developer.apple.com/library/archive/documentation/ToolsLanguages/Conceptual/Xcode_Overview/MeasuringPerformance.html

Also you could remove parts of the app you suspect could be the cause....keep removing until you see a difference.

**Reply by i-mage** [reply]
Thanks for your suggestions!

So, I think the culprit is the  Web Audio API since it's totally jagged in the production mode. During the development mode, everything is smooth and fine. Any experience with this?

This is really frustrating because in the dev mode everything runs absolutely smoothly.

**Comment by Davidlishi** [comment]
> I am developing an app which uses combination of sounds / animations & everything was great and smooth once developing on the notebook and DEV mode on the phone. However once I have built the app in production mode, suddenly the app started lagging and being unresponsive.
> 
> Could anyone pls help me with how to optimize the app or increase the performance?

Hey, 

We recently encountered similar issues with our client's [Phone case app](https://designmycase.co.uk) . We tried running the app on different devices thinking it may be our older iPhone 6 testing device. It did work better on a later iPhone X but we still were getting a little lag. We found issues with the Xcode Version and had to update to the last stable version after pulling our hair out! Lol

Hope you get it sorted 

Good luck

**Reply by i-mage** [reply]
Yes, all sorted, thanks! :-)

**Comment by oppomobiles** [comment]
If your [Android](https://savingbuyers.com/best-oppo-phones-in-2021-you-must-buy/) apps are not working or loading properly, give these possible fixes a try. We’ll elaborate on each of them individually below.

Roll Back to an Older Version of the App
Uninstall Android System WebView Updates
Update the App
Check for Any New Android Updates
Force-Stop the App
Clear the App’s Cache and Data
Uninstall and Install the App Again
Restart Your Phone
Check Your SD Card (If You Have One)
Reset to Factory Settings

**Reply by i-mage** [reply]
Thanks for the tips!

**Comment by quickbit-pro** [comment]
Hi,
We encounter the same problem, in our case problem was solved with disabling JavaScriptObfuscator in quasar.conf for production build (our app is only for internal use so unObfuscated code does not cause any issue, if your is released to public you can check if you can optimize JavaScriptObfuscator to get better preformance in production build)
